Doc/library/typing.rst

```diff
@@ -578,6 +580,30 @@ These can be used as types in annotations and do not support ``[]``.
    * Every type is compatible with :data:`Any`.
    * :data:`Any` is compatible with every type.
 
+.. data:: LiteralString
+
+   Special type that includes only literal strings. A string
+   literal is compatible with ``LiteralString``, as is another
+   ``LiteralString``, but an object typed as just ``str`` is not.
```

Comment:
@JelleZijlstra  Can we also point out that composing literal strings is fine too?

A one-line example below might be useful too.

Reply:
I didn't follow the PEP closely -- how far does this go? I suppose `"a" + "b"` is good. What about `"a" * 3`? Or `"a %s z" % "qqq"`? Or `",".join(("x", "y", "z"))`?

Reply:
All three are fine. Basically, if all the inputs (including `self`) are `LiteralString`, then the output type is a `LiteralString`. (We [listed](https://peps.python.org/pep-0675/#appendix-c-str-methods-that-preserve-literalstring) `str` operations that preserve the `LiteralString` type.)

Reply:
Added

Reply:
@pradeep90 any other feedback? Thanks for the review!